WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationProtobufMessageProcessor.swift

```diff
@@ -0,0 +1,174 @@
+        switch content {
+        case .lastRead:
+
+            await messageLocalStore.updateLastReadMessageTimestamp(
+                message.lastRead,
+                in: conversation
+            )
+
+        case .cleared:
+
+            await messageLocalStore.updateClearedMessageTimestamp(
+                message.cleared,
+                in: conversation
+            )
+
+        case .hidden:
+
+            await messageLocalStore.deleteMessageForSelf(
+                message.hidden,
+                in: conversation
+            )
+
+        case .dataTransfer(let dataTransfer):
+            guard let trackingIdentifier = dataTransfer.trackingIdentifierData else {
+                break
+            }
+
+            await userLocalStore.updateSelfUserAnalyticsID(
+                analyticsID: trackingIdentifier,
+                conversation: conversation
+            )
+
+        case .deleted:
+
+            await messageLocalStore.deleteMessageForEveryone(
+                message.deleted,
+                in: conversation,
+                senderID: senderID.uuid
+            )
+
+        case .reaction:
+
+            await messageLocalStore.addMessageReaction(
+                message.reaction,
+                in: conversation,
+                senderID: senderID.uuid,
+                date: date
+            )
+
+        case .confirmation:
+
+            // Some logic was done here but it seems unnecessary - see legacy `ZMOTRMessage+UpdateEvent`
+            break
+
+        case .buttonActionConfirmation:
+
+            await messageLocalStore.updateButtonStates(
+                message.buttonActionConfirmation,
+                in: conversation
+            )
+
+        case .edited:
+
+            await messageLocalStore.editMessage(
+                message.edited,
+                in: conversation,
+                senderID: senderID.uuid,
+                genericMessage: message,
+                date: date
+            )
+
+        case .clientAction(.resetSession):
+
+            guard let senderClientID else {
+                return WireLogger.eventProcessing.warn(
+                    "clientAction resetSession did not create any message",
+                    attributes: logAttributes
+                )
+            }
+
+            let systemMessageType: SystemMessageType = .sessionReset(
+                sender: (senderID.uuid, senderID.domain),
+                senderClientID: senderClientID,
+                date: date
+            )
+
+            await messageLocalStore.addSystemMessage(
+                messageType: systemMessageType,
+                conversationID: conversationID.uuid,
+                conversationDomain: conversationID.domain
+            )
+
+        case .calling, .availability:
+
+            // cases not handled
+            break
+
+        default:
+
+            await messageLocalStore.addTextMessage(
```

Comment:
The default case doesn't necessarily mean it's a text message. This is especially true for any new message type that is added and falls into the default case. I know it's a lot of work, but it'd be good to also have an exhaustive switch where it's clear how each message type is handled. This means cleaving apart the various ways we handle different generic messages from inside the message local store.

Reply:
I revisited the implementation in https://github.com/wireapp/wire-ios/pull/2164/commits/244f4d59aa504d7edb51a223ddcf618e4d691b75 so we're more transparent on what is done in here. there is definitely still some room for improvement in terms of refactoring specifically getting rid of all the generic message helper methods (that gives a lot of opaqueness) so we treat each content in a more straightforward way. This involves manipulating code that is still used by legacy code. I think it's worth creating a follow-up ticket on this quite large topic that we could treat later on. What do you think ? @johnxnguyen  @netbe

Reply:
Yes let's revise in a separate PR.